### ISSUE TITLE
🐛 Bug Report — Runtime APIs

### ISSUE THREAD

**Issue by KimPlaybit** [cloudflare/workerd#2735]
**What would you like?**

Run a web application with Workerd on a computer with Win32-Arm64 OS.

**Why is this needed?**
With Asus recently releasing laptops with Qualcomm CPUs, this will be more and more requested.
See The Asus Vivobook S 15.

**Other**
I managed to start my Web app with workerd regardless.

By adding in "win32 arm64 LE": "@cloudflare/workerd-windows-64" in Install.js as a knownPackage. I dont know what side-effects this brings.

**Comment by alduboishc** [comment]
That would be nice.  I had to do the same workaround as OP to get my dev env. running.

**Comment by p-romain** [comment]
I was not able to connect to a pg db with workerd for windows x64, under windows arm64. I'm using prisma + hono + wrangler@3 and the request was running indefinitely without error message or timeout. Only solution found was to rollback on wrangler@2. Just for your information if somebody has the same behavior.

**Comment by rafaelmagalhaes** [comment]
any updates on this ?

**Comment by a-type** [comment]
Latest Surface Pro / Laptop line uses Snapdragon X ARM64 processors, continuing the Windows ARM trend. As an owner of one, this would be really convenient. I was intending to evaluate using Durable Objects for a new project, but even the small friction of setting up a WSL environment just to run the tutorial has me putting it off.

**Comment by Badbird5907** [comment]
Hello, any updates on this?
I have a Snapdragon X Elite ARM64 laptop and am unable to run wrangler natively and must use WSL.

**Comment by a-type** [comment]
I recommend anyone watching this, go upstream and follow the roadmap task to add free-tier public Windows ARM64 images to Github Runners. My understanding is that this is blocking Bazel from support, which in turn blocks this project.

https://github.com/github/roadmap/issues/1098

**Comment by a-type** [comment]
Just in, at least the first upstream prerequisite appears to be unblocked: https://github.com/github/roadmap/issues/1098 (public ARM64 Windows action runners) is now shipped.

Not totally familiar with the chain of events here, but I gather Cloudflare may still have to wait on changes in Bazel to support ARM on Windows. Still, this seemed to be the largest blocker.

**Comment by a-type** [comment]
After trying to track down remaining blockers here, it's not clear what remains to be done. https://github.com/cloudflare/workerd/pull/2997 was an early attempt at Windows ARM support stymied by lack of public Arm64 workers and some kind of Bazel incompatibility, but Bazel seems to offically support Win Arm 64 now and the public workers are launched. I wonder if that same changeset could work now. Can anyone more familiar with the requirements in `workerd` comment on whose court the ball is in now, so to speak?

**Comment by fhanau** [comment]
> After trying to track down remaining blockers here, it's not clear what remains to be done. [#2997](https://github.com/cloudflare/workerd/pull/2997) was an early attempt at Windows ARM support stymied by lack of public Arm64 workers and some kind of Bazel incompatibility, but Bazel seems to offically support Win Arm 64 now and the public workers are launched. I wonder if that same changeset could work now. Can anyone more familiar with the requirements in `workerd` comment on whose court the ball is in now, so to speak?

Unfortunately this is still not easy to support, even with the arm64 runner being available based on workerd's dependencies. The build uses rules_js and by extension yq to import npm packages, which is not yet available for arm64 Windows (https://github.com/mikefarah/yq/issues/2348). There might be more similar dependency issues, this is just the first error causing the build to fail. I hope that supporting arm64 will become easier as it becomes available for dependencies.

**Comment by ryanlua** [comment]
Any updates on this? It looks like https://github.com/mikefarah/yq/issues/2348 is resolved, which was the blocker preventing progress on this issue.

**Comment by fhanau** [comment]
> Any updates on this? It looks like [mikefarah/yq#2348](https://github.com/mikefarah/yq/issues/2348) is resolved, which was the blocker preventing progress on this issue.

Unfortunately this is not a priority for the upcoming quarter. It's good that the yq issue is resolved, but it is likely that there are several other dependencies that do not support arm64 Windows/would require complex workarounds.

**Comment by kuitog** [comment]
Not sure if it can be useful for anyone: I am using wrangler with npm and installing nodejs for x64 instead of ARM64 fixed the issue.
Cloudflare sees it as a normal x64 environment.
I do the same with python to fix some dependencies that refuse to work in ARM64 win

**Comment by Master-Hash** [comment]
> By adding in "win32 arm64 LE": "@cloudflare/workerd-windows-64" in Install.js as a knownPackage. I dont know what side-effects this brings.

Before Officially support, doing this in upstream is beneficial.

**Comment by honeymaro** [comment]
any updates?

**Comment by dhruv-m1** [comment]
+1 for this - Windows on ARM is now going mainstream.

Factors highlighted by @a-type, seem to be resolved.

**Comment by piffie** [comment]
Would love if this is starting to work - don't want to force all my other apps t use x64 emulation for node...

**Comment by mbwmbw1337** [comment]
Really need a solution on this... Any updates?

**Comment by Badbird5907** [comment]
> Really need a solution on this... Any updates?

I know it was mentioned a bit earlier in the thread, but i have this patch set up to run workerd under emulation, which works just fine
https://github.com/Badbird5907/silo/blob/master/patches/workerd.patch

**Comment by mbwmbw1337** [comment]
> > Really need a solution on this... Any updates?
> 
> I know it was mentioned a bit earlier in the thread, but i have this patch set up to run workerd under emulation, which works just fine
> https://github.com/Badbird5907/silo/blob/master/patches/workerd.patch

Thank you! This worked flawlessly. Not sure why it's not integrated into the upstream repo. Perhaps submit a PR?

**Comment by danlapid** [comment]
Thanks, we'll be upstreaming a version of the patch above to workerd main

**Comment by mbwmbw1337** [comment]
Fantastic news!

**Comment by Badbird5907** [comment]
> Thanks, we'll be upstreaming a version of the patch above to workerd main

Thanks! I've gone ahead and opened #6882 in hopes we can get it merged while we wait for #6486 
Hope to see it merged soon!
